plugins/content_distribution/lib/kContentDistributionFlowManager.php

```diff
@@ -1617,6 +1623,13 @@ public static function onEntryDeleted(entry $entry)
 				KalturaLog::log("Entry distribution [" . $entryDistribution->getId() . "] already flagged for deletion");
 				continue;
 			}
+			
+			if (!in_array ($entryDistribution->getStatus(), $validStatus))
+			{
+				$entryDistribution->setStatus(EntryDistributionStatus::DELETED);
+				$entryDistribution->setDirtyStatus(EntryDistributionDirtyStatus::NONE);
+				$entryDistribution->save ();
```

Comment:
@tan-tan-kanarek  That is precisely why I added the 'validStatuses' array above and only delete the entry distribution if its status is not one of these.